### PR TITLE
Add hook to verify the peer certiticate.

### PR DESCRIPTION
**Pull request by jyao1**
REF: https://github.com/DMTF/libspdm/issues/148

Signed-off-by: Jiewen Yao <jiewen.yao@intel.com>

### CODE REVIEW

File: include/library/spdm_common_lib.h [hunk 1]
```diff
@@ -447,6 +447,49 @@ void spdm_register_transport_layer_func(
...
+
+  The function shall check the negotiated hash algorithm to check root cert hash.
+  The function shall check the negotiated (req) asym algorithm to determine if it is right cert chain.
+
+  The function returns error if either of above is not satisfied.
+
+  @param  spdm_context                  A pointer to the SPDM context.
+  @param  slot_id                       The number of slot for the certificate chain.
+  @param  cert_chain_size               Indicate the size in bytes of the certificate chain.
+  @param  cert_chain                    A pointer to the buffer storing the certificate chain
```
Comment (steven-bellock) [inline include/library/spdm_common_lib.h:465]: How should this behave if the Responder does not return the root certificate? Would the caller have to append the root certificate to the beginning of the certificate chain?

Comment (jyao1) [reply include/library/spdm_common_lib.h:465]: This cert_chain contains everything returned from responder, including root_cert hash.

The verifier can determine which root_cert should be used to verify the chain, by comparing root_cert hash.

libspdm does not the trust anchor. Only verifier knows the trust anchor.

Comment (steven-bellock) [reply include/library/spdm_common_lib.h:465]: Ok. Understood.

**Comment by steven-bellock** [comment]
Looks good to me.

**Pull request merged**